src/js_native_api_v8.cc

```diff
@@ -893,6 +893,28 @@ napi_status napi_get_property_names(napi_env env,
   return GET_RETURN_STATUS(env);
 }
 
+NAPI_EXTERN napi_status napi_get_own_property_names(napi_env env,
+                                                    napi_value object,
+                                                    napi_value* result) {
+  NAPI_PREAMBLE(env);
+  CHECK_ARG(env, result);
+
+  v8::Local<v8::Context> context = env->context();
+  v8::Local<v8::Object> obj;
+  CHECK_TO_OBJECT(env, context, obj, object);
+
+  v8::MaybeLocal<v8::Array> maybe_own_propertynames = obj->GetOwnPropertyNames(
+      context,
+      static_cast<v8::PropertyFilter>(v8::PropertyFilter::SKIP_SYMBOLS),
+      v8::KeyConversionMode::kConvertToString);
+
+  CHECK_MAYBE_EMPTY(env, maybe_own_propertynames, napi_generic_failure);
```

Comment:
We actually need a variant of this macro that takes into account the fact that this is in a `TryCatch` block:

```C++
#define RETURN_STATUS_IF_FALSE_WITH_PREAMBLE(env, condition, status)  \
  do {                                                                \
    if (!(condition)) {                                               \
      return napi_set_last_error((env),                               \
          try_catch.HasCaught() ? napi_pending_exception : (status)); \
    }                                                                 \
  while (0)
#define CHECK_MAYBE_EMPTY_WITH_PREAMBLE(env, maybe, status) \
  RETURN_STATUS_IF_FALSE_WITH_PREAMBLE((env), !((maybe).IsEmpty()), (status))
```

Could you please add these two macros to `src/js_native_api_v8.h` and then use `CHECK_MAYBE_EMPTY_WITH_PREAMBLE()` here?

Reply:
`NAPI_PREAMBLE` should have done this?

Reply:
ok i will